mobile-app/app/screens/AppNavigator/screens/Balances/components/CreateOrEditAddressLabelForm.tsx

```diff
@@ -28,57 +36,127 @@ export const CreateOrEditAddressLabelForm = memo(({ route, navigation }: Props):
   const {
     address,
     addressLabel,
-    // type,
+    type,
     index,
-    onSubmitButtonPress
+    onSaveButtonPress
   } = route.params
   const [labelInput, setLabelInput] = useState(addressLabel?.label)
+  const [addressInput, setAddressInput] = useState<string | undefined>()
   const bottomSheetComponents = {
     mobile: BottomSheetScrollView,
     web: ThemedScrollView
   }
   const ScrollView = Platform.OS === 'web' ? bottomSheetComponents.web : bottomSheetComponents.mobile
-
-  const [inputErrorMessage, setInputErrorMessage] = useState('')
-  const validateInput = (input: string): boolean => {
+  const { networkName } = useNetworkContext()
+  const addressBook = useSelector((state: RootState) => state.userPreferences.addressBook)
+  const [labelInputErrorMessage, setLabelInputErrorMessage] = useState('')
+  const [addressInputErrorMessage, setAddressInputErrorMessage] = useState('')
+  const validateLabelInput = (input: string): boolean => {
     if (input !== undefined && input.trim().length > 30) {
-      setInputErrorMessage('Address label is too long (max 30 characters)')
+      setLabelInputErrorMessage('Address label is too long (max 30 characters)')
+      return false
+    }
+    if (type === 'create' && input.trim() === '') {
+      setLabelInputErrorMessage('Please enter an address label')
+      return false
+    }
+    setLabelInputErrorMessage('')
+    return true
+  }
+
+  const validateAddressInput = (input: string): boolean => {
+    const decodedAddress = fromAddress(input, networkName)
+    if (decodedAddress === undefined) {
+      setAddressInputErrorMessage('Please enter a valid address')
+      return false
+    }
+    if (addressBook?.[input.trim()] !== undefined) {
+      setAddressInputErrorMessage('Please enter a unique address')
```

Comment:
I assume this one is if an address is existing? We can use a different copy like "This address already exist on your address book". "Unique address" doesn't really mean anything

Reply:
`Address already existed in your address book, please enter a different address`?

Reply:
Address already exist in your address book, please enter a different address.